sfall/LoadGameHook.cpp

```diff
@@ -276,6 +277,45 @@ static void __declspec(naked) LoadGame() {
 		retn;
 	}
 }
+
+void ReadExtraGameMsgFiles()
+{
+	int read;
+	std::string names;
+
+	names.resize(256);
+
+	while ((read = GetPrivateProfileStringA("Misc", "ExtraGameMsgFileList", "",
+		(LPSTR)names.data(), names.size(), ".\\ddraw.ini")) == names.size() - 1)
+		names.resize(names.size() + 256);
+
+	if (names.empty())
+		return;
+
+	names.resize(names.find_first_of('\0'));
+	names.append(",");
+
+	int begin = 0;
+	int end;
+	int length;
+
+	while ((end = names.find_first_of(',', begin)) != std::string::npos)
+	{
+		length = end - begin;
+
+		if (length > 0)
+		{
+			std::string path = "game\\" + names.substr(begin, length) + ".msg";
```

Comment:
Happy to learn that you managed to compile it!

Sure. Sorry I didn't include the instructions earlier.

1. Add a **ExtraGameMsgFileList** key in **Misc** section of ddraw.ini.
2. Assign a comma separated file names without extensions to it. That would be msg files located in \data\text\english\game directory. They will be read and have consecutive ids assigned beginning from 0x2000. I used 0x2000 for consistency since ids beginning from 0x1000 are used for proto files, but it can be changed if needed.

Thus for **ExtraGameMsgFileList=foo,bar** foo.msg will be associated with 0x2000 and bar.msg with 0x2001.

3. Use message_str_game(id, number) to read the messages.